### ISSUE TITLE
Form: show character count on the "title" field

### ISSUE THREAD

**Issue by kristinashu** [mozilla/network-pulse#493]
Title has a character count limit but we don't give users any feedback on this. (screen cap from production site)

<img width="758" alt="screen shot 2017-04-17 at 10 49 20 am" src="https://cloud.githubusercontent.com/assets/1068783/25098113/3cd0f576-235c-11e7-801c-45a6683a4f2c.png">

We should show the character count as they type (like on Twitter or in a google form). It is mocked up in redpen https://redpen.io/ov0d7f8ba9c437ee56

<img width="719" alt="screen shot 2017-04-17 at 10 50 18 am" src="https://cloud.githubusercontent.com/assets/1068783/25098106/38ef57a4-235c-11e7-8d49-3497357a0c89.png">

**Comment by kristinashu** [comment]
Character count should show for both the Title field and Description field.

**Comment by kristinashu** [comment]
Once we have added this to the Title field we should remove "Keep it simple. Name it, don't describe it." There is too much text on the form and the character count will keep people in check.

**Comment by kristinashu** [comment]
@mmmavis do you think this would be a lot of work?

@xmatthewx can we put this into next heartbeat?

**Comment by arush0311** [comment]
Can i take this?

**Comment by mmmavis** [comment]
@arush0311 for sure! 😃 ✨ 

@kristinashu @arush0311 work needed to make this happen depends on how we want the counter to work. e.g., start showing the counter as soon as user starts typing in the field? or show counter after user's hit the "submit" button. Either way it will probably involve PRs on the form library we are using: [react-formbuilder](https://www.npmjs.com/package/react-form-builder) and since @Pomax wrote 99.99% of the code there I'll defer the question to him regarding how much work it would need to enable word/character counter.

**Comment by kristinashu** [comment]
Yes, the counter should show as soon as user starts typing in the field (like Twitter).

**Comment by kristinashu** [comment]
Realize twitter only shows the one number. If this is possible it would be great to do the same:

Before typing (see max number of character):
<img width="570" alt="screen shot 2017-08-01 at 1 42 13 pm" src="https://user-images.githubusercontent.com/1068783/28846261-9a0fb3e2-76bf-11e7-808f-60e3d308097a.png">

Starting typing (the number decreases as you type):
<img width="568" alt="screen shot 2017-08-01 at 1 42 30 pm" src="https://user-images.githubusercontent.com/1068783/28846269-a0193830-76bf-11e7-9b46-e94a84f72c4b.png">

Once I'm over the number of characters, the number is red and has a "-" (ignore the red highlight):
<img width="563" alt="screen shot 2017-08-01 at 1 43 16 pm" src="https://user-images.githubusercontent.com/1068783/28846260-9a0daf66-76bf-11e7-93c8-a68ae85bd57f.png">

**Comment by mmmavis** [comment]
Hey @arush0311 are you still interested in working on this? Timeline wise this will need to be landed this heartbeat to unblock https://github.com/mozilla/network-pulse/issues/626 . @Pomax and I have some convo on https://github.com/Pomax/react-formbuilder/issues/30 which you might wanna have a look. Let us know what you think! 😃

**Comment by alanmoo** [comment]
@arush0311 Did you still want to take this one/did you start it?

**Comment by arush0311** [comment]
@alanmoo actually it depends on https://github.com/Pomax/react-formbuilder/pull/46, which is still under review i think

**Comment by mmmavis** [comment]
Hey @arush0311 looks like there are some Pomax's inline comments in that PR that you haven't got a chance to address. Do you need further clarifications for any of them?

**Comment by arush0311** [comment]
yup. https://github.com/Pomax/react-formbuilder/pull/46#discussion_r134312380

**Comment by Pomax** [comment]
react-formbuilder 0.8.0 was pushed with char/word-limit support, unblocking this issue.

**Comment by mmmavis** [comment]
Reopening... 
just realized there are some more steps to do before we can close this ticket

**Comment by Pomax** [comment]
Which fields? Can we set up a small checkbox list that we can tick off?

**Comment by mmmavis** [comment]
I assume we have to show character count on **all** fields that have character limit. I'll update this ticket title and file a new ticket for the rest of the work.

**Comment by mmmavis** [comment]
Followup tickets:

Add form: https://github.com/mozilla/network-pulse/issues/769
Profile form: (I'm filing it now)

**Comment by mmmavis** [comment]
Oops, sorry... have to reopen it again since we haven't verified it on staging. Moving this to the "pending verification on staging" pipeline. @kristinashu or @xmatthewx please close this ticket if things look good on staging.

**Comment by kristinashu** [comment]
Yeah looks good!
